lib/mpl_toolkits/axes_grid1/parasite_axes.py

```diff
@@ -294,23 +274,27 @@ def twin(self, aux_trans=None, axes_class=None):
         While self will have ticks on the left and bottom axis, the returned
         axes will have ticks on the top and right axis.
         """
-        if axes_class is None:
-            axes_class = self._get_base_axes()
-
-        parasite_axes_class = parasite_axes_class_factory(axes_class)
-
         if aux_trans is None:
             aux_trans = mtransforms.IdentityTransform()
-        ax2 = parasite_axes_class(self, aux_trans, viewlim_mode="transform")
-        self.parasites.append(ax2)
-        ax2._remove_method = self._remove_any_twin
-
+        ax = self._add_twin_common(
+            axes_class, aux_transform=aux_trans, viewlim_mode="transform")
         self.axis["top", "right"].set_visible(False)
+        ax.axis["top", "right"].set_visible(True)
+        ax.axis["left", "bottom"].set_visible(False)
+        return ax
 
-        ax2.axis["top", "right"].set_visible(True)
-        ax2.axis["left", "bottom"].set_visible(False)
+    def _add_twin_common(self, axes_class, **kwargs):
```

Comment:
I'd call this

```suggestion
    def _add_twin_axes(self, axes_class, **kwargs):
```
- It actually does all the work of adding the twin axes. Whether it's a "common" method is secondary, and the docstring even indicates all the callers. 
- `_add_twin_axes` is a bit more explicit than `_add_twin`.

Reply:
sure